examples/workflows/script_with_resources.py

```diff
@@ -0,0 +1,10 @@
+from hera.workflows import Resources, Workflow, script
+
+
+@script(resources=Resources(memory_request="5Gi"))
+def task_with_memory_request():
+    print("ok")
+
+
+with Workflow(generate_name="script-with-resources-", entrypoint="s") as w:
```

Comment:
I think this should be 
```suggestion
with Workflow(generate_name="script-with-resources-", entrypoint="task-with-memory-request") as w:
```

`task_with_memory_request(name="s")` sets the name of the independent task/step but in this workflow this does not run as part of a task/step so the template takes the name of the function but parsed for Argo/K8s

I tried the YAML as a test: 
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: script-with-resources-
spec:
  entrypoint: task-with-memory-request
  templates:
  - name: task-with-memory-request
    script:
      command:
      - python
      image: python:3.8
      resources:
        requests:
          memory: 5Gi
      source: |
                import os
                import sys
                sys.path.append(os.getcwd())
                print("ok")
```

Reply:
@samj1912 maybe we should have a check in the workflow that sets the right entrypoint when a single template is added?

Reply:
Oh I see. I've pushed an update.